pkg/operation/botanist/component/kubeapiserver/deployment.go

```diff
@@ -719,3 +725,176 @@ func (k *kubeAPIServer) probeEndpoint(path string) string {
+			out = append(out, "--oidc-signing-algs="+strings.Join(k.values.OIDC.SigningAlgs, ","))
+		}
+
+		for key, value := range k.values.OIDC.RequiredClaims {
+			out = append(out, "--oidc-required-claim="+fmt.Sprintf("%s=%s", key, value))
+		}
+	}
+
+	out = append(out, "--profiling=false")
+	out = append(out, fmt.Sprintf("--proxy-client-cert-file=%s/%s", volumeMountPathKubeAggregator, secrets.ControlPlaneSecretDataKeyCertificatePEM(SecretNameKubeAggregator)))
+	out = append(out, fmt.Sprintf("--proxy-client-key-file=%s/%s", volumeMountPathKubeAggregator, secrets.ControlPlaneSecretDataKeyPrivateKey(SecretNameKubeAggregator)))
+	out = append(out, fmt.Sprintf("--requestheader-client-ca-file=%s/%s", volumeMountPathCAFrontProxy, secrets.DataKeyCertificateCA))
+	out = append(out, "--requestheader-extra-headers-prefix=X-Remote-Extra-")
+	out = append(out, "--requestheader-group-headers=X-Remote-Group")
+	out = append(out, "--requestheader-username-headers=X-Remote-User")
+
+	if k.values.RuntimeConfig != nil {
+		out = append(out, kutil.MapStringBoolToCommandLineParameter(k.values.RuntimeConfig, "--runtime-config="))
+	}
+
+	if k.values.ServiceAccount.SigningKey != nil {
+		out = append(out, fmt.Sprintf("--service-account-signing-key-file=%s/%s", volumeMountPathServiceAccountSigningKey, SecretServiceAccountSigningKeyDataKeySigningKey))
+		out = append(out, fmt.Sprintf("--service-account-key-file=%s/%s", volumeMountPathServiceAccountSigningKey, SecretServiceAccountSigningKeyDataKeySigningKey))
+	} else {
+		out = append(out, fmt.Sprintf("--service-account-signing-key-file=%s/%s", volumeMountPathServiceAccountKey, secrets.DataKeyRSAPrivateKey))
+		out = append(out, fmt.Sprintf("--service-account-key-file=%s/%s", volumeMountPathServiceAccountKey, secrets.DataKeyRSAPrivateKey))
+	}
+
+	out = append(out, "--service-account-issuer="+k.values.ServiceAccount.Issuer)
+	out = append(out, fmt.Sprintf("--service-cluster-ip-range=%s", k.values.VPN.ServiceNetworkCIDR))
+	out = append(out, fmt.Sprintf("--secure-port=%d", Port))
+	out = append(out, fmt.Sprintf("--token-auth-file=%s/%s", volumeMountPathStaticToken, secrets.DataKeyStaticTokenCSV))
+	out = append(out, fmt.Sprintf("--tls-cert-file=%s/%s", volumeMountPathServer, secrets.ControlPlaneSecretDataKeyCertificatePEM(SecretNameServer)))
+	out = append(out, fmt.Sprintf("--tls-private-key-file=%s/%s", volumeMountPathServer, secrets.ControlPlaneSecretDataKeyPrivateKey(SecretNameServer)))
+	out = append(out, "--tls-cipher-suites="+strings.Join(kutil.TLSCipherSuites(k.values.Version), ","))
+	out = append(out, "--v=2")
+
+	if k.values.WatchCacheSizes != nil {
+		if k.values.WatchCacheSizes.Default != nil {
+			out = append(out, fmt.Sprintf("--default-watch-cache-size=%d", *k.values.WatchCacheSizes.Default))
+		}
+
+		if len(k.values.WatchCacheSizes.Resources) > 0 {
+			var sizes []string
+
+			for _, resource := range k.values.WatchCacheSizes.Resources {
+				size := resource.Resource
```

Comment:
We did a lot of inconsistent things in the Helm charts which are now no longer needed. `Resource` is a required field in the API already, so at this point it will be set. The only optional field is `APIGroup`.